### DISCUSSION TITLE
Non destructive re-indexing and a couple other questions

### DISCUSSION THREAD

**Discussion by emelbardis** [nextcloud/recognize#972] [General]
I've done a lot of manual work organizing my photos with recognize and memories.
Does my manual merging and moving to a cluster of faces 'train' the system for future classifications?

Can I have the system take another look at 'unassigned faces' without deleting the existing faces and clusters?

Are these commands non-destructive to my existing classifications and will they help automatically classify/cluster faces that are currently in need of ?
occ recognize:recrawl
occ recognize:classify
occ recognize:cluster-faces

I'd like to get the system to keep trying to classify faces but retain all my current status.

Thanks

**Comment by marcelklehr** [comment]
> Does my manual merging and moving to a cluster of faces 'train' the system for future classifications?

Yes, when you create a cluster the system ideally learns to recognize the person represented by that cluster.

> occ recognize:recrawl

Is destructive.

> occ recognize:classify

Is destructive.

> occ recognize:cluster-faces

Is not destructive. It will bail out if no face detections are left for clustering.

> I'd like to get the system to keep trying to classify faces but retain all my current status.

The system will automatically detect faces in newly added photos and cluster those face detections as needed. In normal operation no manual intervention is necessary.

**Reply by emelbardis** [reply]
Thanks.

> The system will automatically detect faces in newly added photos and cluster those face detections as needed. In normal operation no manual intervention is necessary.

So what to do about the thousands of unassigned faces of which many are images of already clustered faces?

**Reply by marcelklehr** [reply]
You can run occ recognize:cluster-faces to try and assign those to clusters, but they likely won't get new clusters unless you have lots of new photos. You can assign the manually, if you want to.
